### ISSUE TITLE
Package references contains duplicates

### ISSUE THREAD

**Issue by vcaraulean** [atifaziz/LinqPadless#19]
An extract from csx:
``` 
#r "packages\System.ValueTuple.4.3.0\lib\netstandard1.0\System.ValueTuple.dll"
#r "packages\System.ValueTuple.4.3.0\lib\netstandard1.0\System.ValueTuple.dll"
#r "packages\System.ValueTuple.4.3.0\lib\netstandard1.0\System.ValueTuple.dll"
#r "packages\System.ValueTuple.4.3.0\lib\netstandard1.0\System.ValueTuple.dll"
#r "packages\System.Xml.ReaderWriter.4.3.0\lib\net46\System.Xml.ReaderWriter.dll"
#r "packages\System.Xml.XPath.4.3.0\lib\net46\System.Xml.XPath.dll"
#r "packages\System.Xml.XPath.XDocument.4.3.0\lib\net46\System.Xml.XPath.XDocument.dll"
#r "packages\System.Xml.XPath.XDocument.4.3.0\lib\net46\System.Xml.XPath.XDocument.dll"
```
It might look harmless. For traditional targets (like `net461`) where number of potential references is low there is little overhead. For netstandard targets reference lookup takes 10-20 seconds (running in console), ending up with a list of over 300 lines for the example from #13, lots of duplicates.

Some caching/tracking of "seen" items is probably welcome. Process will be faster and will contain no duplicates.

**Comment by atifaziz** [comment]
Closed by 185713c37eede26c91881d6a0d0524c6e31fff8b

**Comment by atifaziz** [comment]
> Some caching/tracking of "seen" items is probably welcome. Process will be faster and will contain no duplicates.

This is a good idea but probably needs to be tracked as a separate issue.
